test/lib/common.bash

```diff
@@ -227,8 +228,27 @@ checkPackage() {
 	local pkgver=$3
 
 	svn up -q "${TMP}/svn-packages-copy/${pkgbase}"
-	# TODO: Does not fail if one arch is missing
-	compgen -G "${TMP}/svn-packages-copy/${pkgbase}/repos/${repo}-*" >/dev/null
+
+	local dirarches=() pkgbuildarches=()
+	local pkgbuild dirarch pkgbuildver
+	for pkgbuild in "${TMP}/svn-packages-copy/${pkgbase}/repos/${repo}-"+([^-])"/PKGBUILD"; do
```

Comment:
You mention a fear of regular globs not correctly handling $repo with a - in it... I guess the other solution would be to match on ${ARCHES[@]}.